### PR TITLE
/api/articles/:article_id #14

### PR DESCRIPTION
**Pull request by duncan-s-white**
This adds the following for the /api/articles/:article_id endpoint, ticket #14:

- Test for 200 that an object with the correct properties is being returned for a valid id
- Test for 400, bad request if the id is not a valid format
- Test for 404, not found in the database
- Code added to satisfy all the tests

### CODE REVIEW

File: controllers/errors.js [hunk 1]
```diff
@@ -1,3 +1,16 @@
 exports.invalidPath = (req, res) => {
   res.status(404).send({ msg: "Path not found!" });
 };
+
+exports.handleCustomErrors = (err, req, res, next) => {
+  console.log("inside handleCustomErrors");
```
Comment (emilyb93) [inline controllers/errors.js]: Remove console logs once the code is completed and you are no longer trying to debug

File: models/articles.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const db = require("../db/connection.js");
+
+exports.selectArticle = (id) => {
+  return db
+    .query(
+      `SELECT * FROM articles 
+      WHERE article_id = $1`,
```
Comment (emilyb93) [inline models/articles.js]: Close your psql strings with a ;

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Just some minor things here but fix these and you're good to merge this :)

File: controllers/errors.js (hunk 1 above)
Comment (duncan-s-white) [reply controllers/errors.js]: Done

File: models/articles.js (hunk 2 above)
Comment (duncan-s-white) [reply models/articles.js]: Done, thanks!

**Pull request merged**